packages/angular_devkit/build_angular/src/tools/esbuild/angular/compiler-plugin.ts

```diff
@@ -358,10 +358,12 @@ export function createCompilerPlugin(
           };
         } else if (typeof contents === 'string') {
           // A string indicates untransformed output from the TS/NG compiler
+          const sideEffects = await hasSideEffects(args.path);
```

Comment:
```suggestion
          const sideEffects = await hasSideEffects(request);
```
The file path may have been changed by a `fileReplacements` entry.